packages/hooks/use-throttle-render/index.ts

```diff
@@ -1,31 +1,53 @@
 import { onMounted, ref, watch } from 'vue'
+import { isNumber, isObject } from '@element-plus/utils'
 
+import type { SkeletonThrottle } from '@element-plus/components'
 import type { Ref } from 'vue'
 
-export const useThrottleRender = (loading: Ref<boolean>, throttle = 0) => {
+export const useThrottleRender = (
+  loading: Ref<boolean>,
+  throttle: SkeletonThrottle = 0
+) => {
   if (throttle === 0) return loading
-  const throttled = ref(false)
+  const initVal = isObject(throttle) ? throttle.initVal ?? false : false
+  const throttled = ref(initVal)
   let timeoutHandle: ReturnType<typeof setTimeout> | null = null
 
-  const dispatchThrottling = () => {
+  const dispatchThrottling = (timer: number) => {
     if (timeoutHandle) {
       clearTimeout(timeoutHandle)
     }
     timeoutHandle = setTimeout(() => {
       throttled.value = loading.value
-    }, throttle)
+    }, timer)
   }
-  onMounted(dispatchThrottling)
+
+  const dispatcher = (type: 'leading' | 'trailing') => {
+    if (type === 'leading') {
+      if (isNumber(throttle)) {
+        dispatchThrottling(throttle)
+      } else {
+        if (throttle?.leading && throttle.leading > 0) {
+          dispatchThrottling(throttle.leading!)
+        }
+      }
+    } else {
+      if (isObject(throttle) && throttle?.trailing && throttle.trailing > 0) {
```

Comment:
```suggestion
      if (isObject(throttle)) {
```

Reply:
I think `isObject(throttle) && throttle?.trailing && throttle.trailing > 0` can be changed to `Object(throttle) && throttle?.trailing`. The judgment of `throttle.trailing > 0` is essential, otherwise the parameter `timer` in `dispatchThrottling` may throw an error in `setTimeout`.

Reply:
The second parameter of `setTimeout` will not throw an error even if it accepts `undefined` or a negative number. Of course, you can make a more accurate judgment, but we need to make a backup. When the conditions are not met, perhaps we should give a default value of `0`.

Reply:
I have modified the `dispatcher` method again：
```
const dispatcher = (type: 'leading' | 'trailing') => {
    if (type === 'leading') {
      if (isNumber(throttle)) {
        dispatchThrottling(throttle)
      } else {
        dispatchThrottling(throttle.leading)
      }
    } else {
      if (isObject(throttle)) {
        dispatchThrottling(throttle.trailing)
      } else {
        throttled.value = false
      }
    }
  }
```
Then `dispatchThrottling` was modified, adding handling for the `undefined` case as well ：
```
const dispatchThrottling = (timer: number | undefined) => {
    if (isUndefined(timer)) {
      throttled.value = loading.value
      return
    }
    if (timeoutHandle) {
      clearTimeout(timeoutHandle)
    }
    timeoutHandle = setTimeout(() => {
      throttled.value = loading.value
    }, timer)
  }
```